core/partitioning/partitioningctx/PartitioningCtx.cpp

```diff
@@ -15,7 +15,8 @@ PartitioningCtx::PartitioningCtx(torch::jit::Block* b, PartitioningInfo info)
 }
 
 void PartitioningCtx::_load_nodes_into_decision_map(torch::jit::Block* b) {
-  if (b->owningNode() && b->owningNode()->kind() == torch::jit::prim::Loop)
+  // won't load nodes if these nodes are in prim::loop or if these nodes are 2-level nested
+  if (b->owningNode() && (b->owningNode()->kind() == torch::jit::prim::Loop || b->owningNode()->owningBlock()->owningNode()))
```

Comment:
Are there ways that `b->owningNode()->owningBlock()->owningNode()` could cause a segfault? For example, I am wondering about the case where the whole graph is a single block, then would `b->owningNode()->owningBlock()` be a `nullptr`?

Reply:
It won't. 
In that case, `b->owningNode()` is `null` and it won't check the rest part of that if statement.